manifests/services/postgresql/keepalived.pp

```diff
@@ -1,17 +1,28 @@
 # Configuring keepalived for the postgres IP
 class profile::services::postgresql::keepalived {
-  $vrrp_password = hiera('profile::keepalived::vrrp_password')
+  $vrrp_password = lookup('profile::keepalived::vrrp_password', String)
 
-  $v4ip = hiera('profile::postgres::ipv4')
-  $v4id = hiera('profile::postgres::ipv4::id')
-  $v4pri = hiera('profile::postgres::ipv4::priority')
-  $v6ip = hiera('profile::postgres::ipv6', false)
-  $v6id = hiera('profile::postgres::ipv6::id', false)
-  $v6pri = hiera('profile::postgres::ipv6::priority', false)
-
-  $management_if = hiera('profile::interfaces::management')
+  $v4ip = lookup('profile::postgres::ipv4', Stdlib::IP::Address::V4)
+  $v4id = lookup('profile::postgres::ipv4::id', Integer)
+  $v4pri = lookup('profile::postgres::ipv4::priority', Integer)
+  $v6ip = lookup('profile::postgres::ipv6', {
+    'value_type'    => Variant[Stdlib::IP::Address::V6, Boolean],
+    'default_value' => false
+    })
+  $v6id = lookup('profile::postgres::ipv6::id', {
+    'value_type'    => Variant[Integer, Boolean],
+    'default_value' => false
```

Comment:
Denne må også være et tall; men det må være unikt for hver instans i et gitt subnett. Så; enten må vi bare sette et tall; eller så må vi sette en error om denne er false samtidig som at $v6ip ikke er false.

Reply:
Her tenker jeg vi ikke trenger noen default? Altså at puppet selv bare kaster en error om at du har glemt å sette en verdi i hiera?

Reply:
Eller. Nvm.. Kanskje vi bare setter et tall ja. Ellers blir det mye "hvis om atte dersom"-sjekker..